### PR TITLE
feat(frontend): create auth wrapper components bb-189

### PR DESCRIPTION
**Pull request by iqbalalayubbi**
- Make the **auth wrapper** component and render **header**, **sidebar**, and **quiz** inside
- Pass the component into a protected route component as a prop

### CODE REVIEW

File: apps/frontend/src/index.tsx [hunk 1]
```diff
@@ -24,7 +26,7 @@ createRoot(document.querySelector("#root") as HTMLElement).render(
 							{
 								element: (
 									<ProtectedRoute
-										component="Root"
+										component={<AuthWrapper />}
```
Comment (fshabanov) [inline apps/frontend/src/index.tsx:29]: Should be used inside ProtectedRoute, it should wrap the `component` inside of it

File: apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import { Header, Sidebar } from "../components.js";
+
+const AuthWrapper: React.FC = () => {
+	return (
+		<>
+			<Header />
+			<Sidebar />
+			<QuizForm />
+		</>
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx:11]: Make sure to keep styling, add a screenshot to this PR

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx (hunk 2 above)
Comment (iqbalalayubbi) [reply apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx:11]: the root page look like this

Screenshot from 2024-08-26 20-51-52

File: apps/frontend/src/index.tsx [hunk 3]
```diff
@@ -22,12 +22,7 @@ createRoot(document.querySelector("#root") as HTMLElement).render(
 					{
 						children: [
 							{
-								element: (
-									<ProtectedRoute
-										component="Root"
-										redirectTo={AppRoute.SIGN_IN}
-									/>
-								),
+								element: <ProtectedRoute redirectTo={AppRoute.SIGN_IN} />,
```
Comment (fshabanov) [inline apps/frontend/src/index.tsx:25]: Return back the change (we should show component)

File: apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx (hunk 2 above)
Comment (fshabanov) [reply apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx:11]: The layout is not correct. Sidebar should be on the left side

File: apps/frontend/src/libs/components/protected-route/protected-route.tsx [hunk 4]
```diff
@@ -3,19 +3,17 @@ import { type AppRoute } from "~/libs/enums/enums.js";
...
-const ProtectedRoute: React.FC<Properties> = ({
-	component,
-	redirectTo,
-}: Properties) => {
+const ProtectedRoute: React.FC<Properties> = ({ redirectTo }: Properties) => {
 	const user = useAppSelector(({ auth }) => auth.user);
 
 	if (user) {
-		return component;
+		return <AuthWrapper />;
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/protected-route/protected-route.tsx:16]:
```suggestion
		return <AuthWrapper>{component}</AuthWrapper>
```

**Review by fshabanov — CHANGES_REQUESTED** [review]

File: apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx (hunk 2 above)
Comment (iqbalalayubbi) [reply apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx]: i try to make the style for auth wrapper and look like this

Screenshot from 2024-08-26 22-16-24

File: apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx [hunk 5]
```diff
@@ -0,0 +1,24 @@
+// import { QuizForm } from "~/pages/quiz/libs/components/quiz-form/quiz-form.jsx";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx:1]:
```suggestion
```

File: packages/shared/CHANGELOG.md [hunk 6]
```diff
@@ -36,26 +36,68 @@
 * authorization token (JWT) bb-10 ([#40](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/issues/40)) ([16e3c35](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/commit/16e3c353ff700ab27a7b6af4fa7b3c17059cc916))
 
 ## [1.2.0](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/compare/shared-v1.1.0...shared-v1.2.0) (2024-08-21)
+>>>>>>> f9749449ad8a29e69c8f5599fce32330826a92fd
```
Comment (fshabanov) [inline packages/shared/CHANGELOG.md:39]: You should have no changes in this file

File: apps/frontend/src/index.tsx [hunk 7]
```diff
@@ -24,7 +25,7 @@ createRoot(document.querySelector("#root") as HTMLElement).render(
 							{
 								element: (
 									<ProtectedRoute
-										component="Root"
+										component={<QuizForm />}
```
Comment (fshabanov) [inline apps/frontend/src/index.tsx:28]: Move `<QuizForm />` to Quiz screen, keep this one as "Root" for now

**Review by fshabanov — CHANGES_REQUESTED** [review]
Add a screenshot with updated layout

File: packages/shared/CHANGELOG.md (hunk 6 above)
Comment (iqbalalayubbi) [reply packages/shared/CHANGELOG.md:39]: i think this change because i merge with main branch

File: apps/frontend/src/index.tsx (hunk 7 above)
Comment (iqbalalayubbi) [reply apps/frontend/src/index.tsx:28]: in this case, i pass the component in app

File: apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx (hunk 5 above)
Comment (iqbalalayubbi) [reply apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx:1]: After change the prop into root. the UI look like this

Screenshot from 2024-08-26 23-08-54

File: apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx [hunk 8]
```diff
@@ -0,0 +1,22 @@
+import { Header, Sidebar } from "../components.js";
```
Comment (fshabanov) [inline apps/frontend/src/libs/components/auth-wrapper.tsx/auth-wrapper.tsx:1]: The folder name should not have `.tsx`

File: packages/shared/CHANGELOG.md [hunk 9]
```diff
@@ -42,20 +42,61 @@
 
 * implement sign-in functionality bb-6 ([#37](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/issues/37)) ([1f2b54c](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/commit/1f2b54c94efa5e1553bc8b4bd24f2fdf2b0f8053))
 
-## [1.1.0](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/compare/shared-v1.0.0...shared-v1.1.0) (2024-08-21)
+## [1.6.0](https://github.com/BinaryStudioAcademy/bsa-2024-bebalance/compare/v1.5.0...v1.6.0) (2024-08-21)
```
Comment (fshabanov) [inline packages/shared/CHANGELOG.md:45]: Make sure you don't have any changes in this file

**Review by fshabanov — CHANGES_REQUESTED** [review]

**Review by fshabanov — APPROVED** [review]

**Pull request merged**